x/leverage/keeper/collateral.go

```diff
@@ -78,9 +78,67 @@ func (k Keeper) GetBorrowerCollateral(ctx sdk.Context, borrowerAddr sdk.AccAddre
 	return totalCollateral
 }
 
-// GetEligibleLiquidationTargets returns a list of borrower addresses eligible for liquidation
+// addressExistInList returns true if address exist in the list of addresses.
+func addressExistInList(list []sdk.AccAddress, addrToCheck sdk.AccAddress) bool {
+	for _, addr := range list {
+		if addrToCheck.Equals(addr) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// GetEligibleLiquidationTargets returns a list of borrower addresses eligible for liquidation.
 func (k Keeper) GetEligibleLiquidationTargets(ctx sdk.Context) ([]sdk.AccAddress, error) {
-	// TODO: Complete this to support liquidation tool
 	// ref: https://github.com/umee-network/umee/issues/229
-	return nil, nil
+	store := ctx.KVStore(k.storeKey)
+	borrowPrefix := types.CreateLoanKeyNoAddress()
+
+	iter := sdk.KVStorePrefixIterator(store, borrowPrefix)
+	defer iter.Close()
+
+	addressEligibleToLiquidation := []sdk.AccAddress{}
+
+	// Iterate over all open borrows, accruing interest on each and collecting new
+	// reserves.
+	for ; iter.Valid(); iter.Next() {
+		// key is borrowPrefix | lengthPrefixed(borrowerAddr) | denom | 0x00
+		key, _ := iter.Key(), iter.Value()
+
+		// remove prefix | denom and null-terminator
+		borrowerAddr := types.GetAddressFromKeyWithPrefix(key, borrowPrefix)
+
+		// if the address is already on the list it can move to the next
+		if addressExistInList(addressEligibleToLiquidation, borrowerAddr) {
+			continue
+		}
+
+		// get total borrowed by borrower (all denoms)
+		borrowed := k.GetBorrowerBorrows(ctx, borrowerAddr)
+
+		// get borrower uToken balances, for all uToken denoms enabled as collateral
+		collateral := k.GetBorrowerCollateral(ctx, borrowerAddr)
+
+		// use oracle helper functions to find total borrowed value in USD
+		borrowValue, err := k.TotalTokenValue(ctx, borrowed)
+		if err != nil {
+			return nil, err
+		}
+
+		// use collateral weights to compute borrow limit from enabled collateral
+		borrowLimit, err := k.CalculateBorrowLimit(ctx, collateral)
+		if err != nil {
+			return nil, err
+		}
+
+		// confirm borrower's eligibility for liquidation
+		if borrowLimit.LT(borrowValue) {
+			continue
+		}
+
+		addressEligibleToLiquidation = append(addressEligibleToLiquidation, borrowerAddr)
```

Comment:
```suggestion
		// check borrower's eligibility for liquidation
		if borrowLimit.LT(borrowValue) {
			addressEligibleToLiquidation = append(addressEligibleToLiquidation, borrowerAddr)
		}
```

Reply:
I changed a little bit to let the append outside of if condition